### PR TITLE
librariesindex: Fix nil pointer. Refs #1176

### PR DESCRIPTION
**Pull request by r10r**
Let the library index return the latest known version,
if a library without a version is updated.

Signed-off-by: Ruben Jenster <r.jenster@drachenfels.de>

### CODE REVIEW

File: arduino/libraries/librariesindex/index.go [hunk 1]
```diff
@@ -136,6 +137,10 @@ func (idx *Index) FindLibraryUpdate(lib *libraries.Library) *Release {
 	if indexLib == nil {
 		return nil
 	}
+	if lib.Version == nil {
+		fmt.Printf("[WARN] version for library loaded from %s is nil\n", lib.InstallDir)
```
Comment (silvanocerza) [inline arduino/libraries/librariesindex/index.go:141]: Remove this line please, printing in this package might cause issues.

**Review by silvanocerza — CHANGES_REQUESTED** [review]
A small change requested. Would be great if you could add an integration test to reproduce the issue so we won't have regressions in the future.

**Comment by r10r** [comment]
> A small change requested. Would be great if you could add an integration test to reproduce the issue so we won't have regressions in the future.

Sure I can - but it may take some time. 
Where should the integration test go to? (Btw. have you seen the test I've added  to `index_test.go`  ?)

File: arduino/libraries/librariesindex/index.go (hunk 1 above)
Comment (r10r) [reply arduino/libraries/librariesindex/index.go]: Done.

**Comment by silvanocerza** [comment]
> Sure I can - but it may take some time.
> Where should the integration test go to?

Integration tests go in the `test` folder in the project root, they're written in Python and call the CLI directly. There's more information in the [official docs](https://arduino.github.io/arduino-cli/latest/CONTRIBUTING/#integration-tests). I'd probably add tests for any command that is affected by this change, `lib install` and `lib list` for sure, I'd have to verify if there are others.
If you want I can take care of it, let me know.

> (Btw. have you seen the test I've added to `index_test.go` ?)

Yep, that's fine for unit tests.

**Comment by r10r** [comment]
> If you want I can take care of it, let me know.

Sorry, but I'm not used to Python, It'll take more time for you to review my code than to write it yourself.
Anyway it's good to know for the future. Thanks!

**Comment by silvanocerza** [comment]
> Sorry, but I'm not used to Python, It'll take more time for you to review my code than to write it yourself.
> Anyway it's good to know for the future. Thanks!

No worries, I'll take care of it as soon as I can, I'll notify you when closing the issue.
Thanks for reporting the issue and providing a fix. :)

**Comment by silvanocerza** [comment]
@r10r I had to cherry pick your commits to a separate branch and create a new PR #1189. It would have been problematic to merge this since it points to your fork's `master` branch and didn't want to put you through the hassle of creating a new PR. 

You'll result as contributor anyway so don't worry about that.

The new PR is #1189, I'll close this one as soon as that is merged in.

**Comment by silvanocerza** [comment]
Closed since #1189 has been merged.

**Comment by r10r** [comment]
I'm really happy to see this merged. Thanks for your support!

**Pull request closed without merging**